module-router.c

```diff
@@ -630,6 +630,62 @@ static void replace_whitespace_with_hash(char* string) {
 }
 
 /**
+ * @brief This function validates the name, if returns false if pointer is
+ * NULL or the name is blank.
+ * @param name: pointer to the name
+ * @ret bool: true if name is valid and vice versa.
+ */
+static bool is_valid_name(const char* name )
+{
+    bool is_valid = true;
+    if((name == NULL) || (name[0] == '\0'))
+    {
+        is_valid = false;
+    }
+    return is_valid;
+}
+
+/**
+ * @brief This function returns the AudioManager name from the pulseaudio
+ * properties list . First check if the media.role property is present and
+ * not blank, then use media.role, Else use application.name.
+ * @param: sink_input: The pointer to the pulseaudio sink_input.
+ *         am_name: The audiomanager name of the sink_input.
+ */
+static void get_am_name_from_media_role(pa_proplist* proplist, char* am_name)
+{
+    char *name_copy = (char*) pa_proplist_gets( proplist,
+                                                PA_PROP_MEDIA_ROLE);
+    if(!is_valid_name(name_copy))
+    {
+        name_copy = (char*) pa_proplist_gets( proplist,
+                                              PA_PROP_APPLICATION_NAME);
+    }
+    if(is_valid_name(name_copy))
+    {
+        strncpy(am_name, name_copy, AM_MAX_NAME_LENGTH);
+        replace_whitespace_with_hash(am_name);
+    }
+}
```

Comment:
Function should be implemented for one purpose.
This means that it should be better to split it to one for media.role and another for media.name.
Ether role or name depends on higher decision, not here.